subsys/logging/log_core.c

```diff
@@ -396,12 +396,24 @@ void log_generic(struct log_msg_ids src_level, const char *fmt, va_list ap)
 	} else {
 		log_arg_t args[LOG_MAX_NARGS];
 		u32_t nargs = count_args(fmt);
+		u32_t mask;
 
 		__ASSERT_NO_MSG(nargs < LOG_MAX_NARGS);
 		for (int i = 0; i < nargs; i++) {
 			args[i] = va_arg(ap, log_arg_t);
 		}
 
+		if (enable_strdup && ( mask = z_log_get_s_mask(fmt, nargs ))) {
```

Comment:
Question: Why is this better than strdupping final string?

Reply:
It is better because:
1) strdup-ed string is shorter
2) we don't process formatting like %d etc when the log is being called, but later, when system works in idle